tensordict/prototype/tensordictclass.py

```diff
@@ -57,6 +57,10 @@ def __init__(self, *args, _tensordict=None, **kwargs):
                 assert isinstance(_tensordict, TensorDict)
                 if args or kwargs:
                     raise ValueError("Cannot pass both args/kwargs and _tensordict.")
+                if not (EXPECTED_KEYS - set(_tensordict.keys()) == {"batch_size"}):
```

Comment:
Say `EXPECTED_KEYS = {'a', 'b', 'c'}` and `set(_tensordict.keys()) = {'a', 'd'}`
then `(EXPECTED_KEYS - set(_tensordict.keys())` is `{'b', 'c'}` but if these keys are optional the real problem should be that there is a `'d'` key that is not expcted

Reply:
Yes, I just spotted that also. It breaks the tests, sorry. Pushing a change to revert the errors I introduced.